google/cloud/spanner_v1/_opentelemetry_tracing.py

```diff
@@ -105,4 +105,10 @@ def trace_call(name, session, extra_attributes=None, observability_options=None)
             span.record_exception(error)
             raise
         else:
-            span.set_status(Status(StatusCode.OK))
```

Comment:
I am a bit confused here. Can you help me understand the scenario where this happens.
As per my understanding,
1. When there is `an exception`, the else block will never be invoked and only the error status is set.
2. When there is `no exception`, then the else block will be invoked and `OK` status is set

Reply:
So if an exception doesn't occur the else block is executed to set to OK, but the code in the with statement can and will change the status for example to ERROR; when it comes to this block then it gets changed back to OK which is not correct. Please see the test I provided in the issue too.

Reply:
Had an offline conversation with @odeke-em .
The scenario here is `When a transaction is aborted and retried, we want to record that transaction status`